src/debugging-support-in-rustc.md

```diff
@@ -268,6 +268,36 @@ Focus is to let macros decide what to do. This can be achieved by having some ki
 that lets the macro tell the compiler where the line marker should be. This affects where you
 set the breakpoints and what happens when you step it.
 
+## Source file checksums in debug info
+
+Both DWARF and CodeView (PDB) support embedding a cryptographic hash of each source file that
+contributed the associated binary.
+
+The cryptographic hash can be used by a debugger to verify that the source file matches the
+executable. If the source file does not match, the debugger can provide a warning to the user.
+
+The hash can also be used to prove that a given source file has not been modified since it was
+used to compile an executable. Because MD5 and SHA1 both have demonstrated vulnerabilities,
+using SHA256 is recommended for this application.
+
+#### DWARF 5
+DWARF version 5 supports embedding an MD5 hash to validate the source file version in use.
+DWARF 5 - Section 6.2.4.1 opcode DW_LNCT_MD5
+
+#### LLVM
+LLVM IR supports MD5, SHA1, and SHA256 source file checksums in the DIFile node.
+
+LLVM DIFile documentation: [https://llvm.org/docs/LangRef.html#difile]
+
+#### Microsoft Visual C++ Compiler /ZH option
+The MSVC compiler supports embedding MD5, SHA1, or SHA256 hashes in the PDB using the `/ZH`
+compiler option.
+
+MSVC /ZH documentation: [https://docs.microsoft.com/en-us/cpp/build/reference/zh]
```

Comment:
Nit: Maybe make this proper links, like so?

[MSVC /ZH documentation](https://docs.microsoft.com/en-us/cpp/build/reference/zh)